### PR TITLE
fix(GUI): make archive-embedded svgs work again

### PR DESCRIPTION
**Pull request by Shou**
We make the svg-icon component accept XML in its path argument to handle
archive-embedded SVG icons.

Changelog-Entry: Make archive-embedded SVG icons work again.
Fixes: https://github.com/resin-io/etcher/issues/1636

### CODE REVIEW

**Comment by lurch** [comment]
Haha, I was just about to leave a comment asking for a unit-test, when you added the second commit to this PR :laughing:

**Comment by jhermsmeier** [comment]
Can confirm it mitigates the little bugger now:
screen shot 2017-07-28 at 16 51 16

File: lib/gui/components/svg-icon.js [hunk 1]
```diff
@@ -53,9 +54,24 @@ class SVGIcon extends react.Component {
     // relatively to the `index.html`.
     const imagePath = path.join(__dirname, this.props.path);
 
-    const contents = fs.readFileSync(imagePath, {
-      encoding: 'utf8'
-    });
+    let contents;
```
Comment (jhermsmeier) [inline lib/gui/components/svg-icon.js]: The linter's yelling about this not being initialized to a value; maybe just set it to an empty string; `let contents = '';`

**Comment by Shou** [comment]
Passes tests and linter locally now, so this is good for reviewing now! Should also repel rickrolls.

**Comment by lurch** [comment]
Just out of curiosity, what happens if it's given a malformed SVG that the DOMParser can't parse correctly?

**Comment by jhermsmeier** [comment]
Then the DOMParser will return an error document: https://developer.mozilla.org/en-US/docs/Web/API/DOMParser#Error_handling – you can also see that if you load up a malformed SVG in your browser, for example.

**Comment by Shou** [comment]
Looks like that causes it to use the previous icon, which in some cases isn't the default, if you've selected another image with an embedded icon, so I need to fix that.

**Comment by jhermsmeier** [comment]
@Shou I just started working on handling parser errors, so don't bother, I'm on it :)

**Comment by lurch** [comment]
Presumably you'll also be checking how it copes with malformed PNGs too? ;-)

**Comment by jhermsmeier** [comment]
Chromium handles those just fine on its own. Also, we don't support custom PNGs – it has to be an SVG, see [`lib/image-stream/archive.js#L112`](https://github.com/resin-io/etcher/blob/master/lib/image-stream/archive.js#L112):

```js
logo: extractEntryByPath(archive, `${basePath}/logo.svg`, options),
```

**Comment by lurch** [comment]
Mea culpa - didn't realise that.

**Pull request merged**